pkg/scheduler/algorithm/predicates/csi_volume_predicate.go

```diff
@@ -169,9 +172,36 @@ func (c *CSIMaxVolumeLimitChecker) getCSIDriver(pvc *v1.PersistentVolumeClaim) (
 
 	csiSource := pv.Spec.PersistentVolumeSource.CSI
 	if csiSource == nil {
-		klog.V(5).Infof("Not considering non-CSI volume %s/%s", namespace, pvcName)
-		return placeHolderCSIDriver, placeHolderHandle
+		if !csilib.IsPVMigratable(pv) {
+			klog.V(5).Infof("Non-CSI volume %s/%s is not migratable", namespace, pvcName)
+			return "", ""
+		}
+
+		pluginName, err := csilib.GetInTreePluginNameFromSpec(pv, nil)
+		if err != nil {
+			klog.V(5).Infof("Unable to look up plugin name from PV spec: %v", err)
+			return "", ""
+		}
+
+		if !isCSIMigrationOn(csiNode, pluginName) {
+			klog.V(5).Infof("CSI Migration of plugin %s is not enabled", pluginName)
+			return "", ""
+		}
+
+		csiPV, err := csilib.TranslateInTreePVToCSI(pv)
+		if err != nil {
+			klog.V(5).Infof("Unable to translate in-tree volume to CSI: %v", err)
+			return "", ""
+		}
+
+		if csiPV.Spec.PersistentVolumeSource.CSI == nil {
+			klog.V(5).Infof("Unable to get a valid volume source for translated PV %s", pvName)
+			return "", ""
+		}
+
+		csiSource = csiPV.Spec.PersistentVolumeSource.CSI
```

Comment:
Do we have a scenario where both CSI and in-tree driver can be in-use on a node? If yes - this code will not count them correctly. cc @davidz627 @jsafrane

Reply:
Do you mean when migration is off and we have the in-tree plugin and its "migration" CSI Driver both be in use on the node?

This is possible.